tests/foreman/cli/test_rhcloud_iop.py

```diff
@@ -0,0 +1,262 @@
+    result = satellite.execute('satellite-maintain health check --label services-up -y')
+    assert result.status == 0, 'Not all services are running'
+
+    org = satellite.api.Organization().create()
+    satellite.upload_manifest(org.id, module_sca_manifest.content)
+
+    activation_key = satellite.api.ActivationKey(
+        content_view=org.default_content_view,
+        organization=org,
+        environment=satellite.api.LifecycleEnvironment(id=org.library.id),
+        service_level='Self-Support',
+        purpose_usage='test-usage',
+        purpose_role='test-role',
+        auto_attach=False,
+    ).create()
+
+    host.configure_rex(satellite=satellite, org=org, register=False)
+    host.configure_insights_client(
+        satellite=satellite,
+        activation_key=activation_key,
+        org=org,
+        rhel_distro=f"rhel{host.os_version.major}",
+    )
+
+    result = host.execute('insights-client')
+    assert result.status == 0, 'insights-client upload failed'
+
+
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('N-0')
+def test_disable_enable_iop(satellite_iop, module_sca_manifest, rhel_contenthost):
+    """Install Satellite + IoP, disable, re-enable.
+
+    :id: abe165e1-a3a4-413d-b6aa-5cb51acfeb2e
+
+    :steps:
+
+        1. Install Satellite and IoP
+        2. Assert all IoP services are running
+        3. Disable IoP by running satellite-installer with `--iop-ensure absent`
+        4. Assert all IoP services are stopped, and podman containers, networks, secrets, and volumes are removed
+        5. Re-enable IoP with `--iop-ensure present`
+        6. Assert all IoP services are running
+
+    :expectedresults: IoP services should be running or absent as configured by the `iop-ensure` installer option
+
+    :CaseAutomation: Automated
+    """
+    satellite = satellite_iop
+    host = rhel_contenthost
+
+    # Register the Insights client
+    org = satellite.api.Organization().create()
+    satellite.upload_manifest(org.id, module_sca_manifest.content)
+
+    activation_key = satellite.api.ActivationKey(
+        content_view=org.default_content_view,
+        organization=org,
+        environment=satellite.api.LifecycleEnvironment(id=org.library.id),
+        service_level='Self-Support',
+        purpose_usage='test-usage',
+        purpose_role='test-role',
+        auto_attach=False,
+    ).create()
+
+    host.configure_rex(satellite=satellite, org=org, register=False)
+    host.configure_insights_client(
+        satellite=satellite,
+        activation_key=activation_key,
+        org=org,
+        rhel_distro=f"rhel{host.os_version.major}",
+    )
+
+    result = host.execute('insights-client')
+    assert result.status == 0, 'Initial insights-client upload failed'
+
+    # Disable IoP
+    command = InstallerCommand(iop_ensure='absent').get_command()
+    result = satellite.execute(command, timeout='10m')
+    assert result.status == 0, 'Failed to disable IoP'
```

Comment:
double check this assertion. Make sure result.status actually returns the status

Reply:
Verified